<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>RecSettingsWindow</class>
 <widget class="QWidget" name="RecSettingsWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>480</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Recording Settings</string>
  </property>
  <property name="styleSheet">
   <string notr="true">
    
    /* === General Styles === */
    
    .QPushButton {
      border: 1px solid black;
      border-top-color: rgb(50,50,50);
      border-left-color: rgb(50,50,50);
      background: white;
      margin: 20px; /*Use margin to increase clickable area. Makes the whole thing much less fiddly.*/
    }
    
    .QPushButton:pressed {
      border: 1px solid rgb(50,50,50);
      border-top-color: black;
      border-left-color: black;
      background: rgb(240,240,240);
    }
    
    .QComboBox {
      /*subcontrol-origin: padding; does nothing but mess up the drop-down button*/
      background: white;
      border: 1px solid black;
      /*margin: 10px; breaks drop-down; hard to fix*/
      padding-left: 10px;
    }
    QComboBox:on {
      /*when dropdown exists*/
    }
    
    .QComboBox QAbstractItemView { /*This is the drop-down menu.*/
      border: 1px solid black;
      color: black;
      selection-background-color: grey;
    }
    .QComboBox QAbstractItemView::item {
      padding: 10px;
    }
    
    QComboBox::drop-down {
      width: 40px;
      border: 0px solid black;
      border-left-width: 1px;
      background: white;
      color: black;
      max-height: 100px;
    }
    QComboBox::down-arrow {
      image: url(assets/wedge-down.png);
    }
    
    
    QSpinBox { 
      border: 1px solid black;
      padding-right: 40px;
      font-size: 6pt;
      margin: 10px;
      padding-left: 10px;
    }  
    QSpinBox::up-button { 
      subcontrol-position: right; 
      right: 40px; 
      image: url(assets/wedge-up.png);
    }
    QSpinBox::down-button { 
      subcontrol-position: right; 
      image: url(assets/wedge-down.png);
    }
    QSpinBox::up-button, QSpinBox::down-button {
      border: 0px solid black;
      border-left-width: 1px;
      width: 40px; 
      height: 40px;
    }
    
    .CamLineEdit {
      border: 1px solid black;
      margin: 10px; /*This doesn't show up in Qt Designer, but it works in the app.*/
      padding-left: 10px;
    }
    
    
    
    /* === Screen-Specific Styles === */
    
    /*
      Open up a window for the camera preview.
      QWidget must be set to transparent background; there's some out-of-tree
      item that's drawing opaque over the preview window.
    */
    
    QWidget {
      background: transparent;
    }
    
    RecSettingsWindow &gt; * {
      background: white;
    }
    
    #frame, #frameImage {
      background: transparent;
      border: 1px solid black;
    }
    
    #passepartoutTop, #passepartoutBottom, #passepartoutLeft, #passepartoutRight {
      background: rgba(127,127,127,64);
      border-width: 0;
      border-radius: 0;
    }
    
    /*For some reason, these two borders are rendered one pixel narrower than they should be. (This seems independant of the set DPI.)*/
    #spinHRes, #spinHOffset {
      border-right-width: 2px;
    }
    #spinVRes, #spinVOffset {
      border-left-width: 2px;
    }
   </string>
  </property>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>20</y>
     <width>131</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>8</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Picture</string>
   </property>
  </widget>
  <widget class="QComboBox" name="comboRes">
   <property name="geometry">
    <rect>
     <x>160</x>
     <y>20</y>
     <width>341</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>6</pointsize>
    </font>
   </property>
   <item>
    <property name="text">
     <string>800×600 @ 2873fps</string>
    </property>
   </item>
  </widget>
  <widget class="CamSpinBox" name="spinHRes">
   <property name="geometry">
    <rect>
     <x>150</x>
     <y>60</y>
     <width>171</width>
     <height>61</height>
    </rect>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="baseSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
   <property name="font">
    <font>
     <pointsize>6</pointsize>
     <weight>50</weight>
     <bold>false</bold>
    </font>
   </property>
  </widget>
  <widget class="CamSpinBox" name="spinVRes">
   <property name="geometry">
    <rect>
     <x>340</x>
     <y>60</y>
     <width>171</width>
     <height>61</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>6</pointsize>
     <weight>50</weight>
     <italic>false</italic>
     <bold>false</bold>
    </font>
   </property>
   <property name="frame">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="CamSpinBox" name="spinHOffset">
   <property name="geometry">
    <rect>
     <x>150</x>
     <y>120</y>
     <width>171</width>
     <height>61</height>
    </rect>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="baseSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
   <property name="font">
    <font>
     <pointsize>6</pointsize>
     <weight>50</weight>
     <italic>false</italic>
     <bold>false</bold>
    </font>
   </property>
  </widget>
  <widget class="CamSpinBox" name="spinVOffset">
   <property name="geometry">
    <rect>
     <x>340</x>
     <y>120</y>
     <width>170</width>
     <height>60</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>6</pointsize>
     <weight>50</weight>
     <italic>false</italic>
     <bold>false</bold>
    </font>
   </property>
   <property name="frame">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>130</y>
     <width>140</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>8</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Offset</string>
   </property>
  </widget>
  <widget class="QPushButton" name="cmdMax">
   <property name="geometry">
    <rect>
     <x>400</x>
     <y>170</y>
     <width>121</width>
     <height>81</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Max</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_3">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>190</y>
     <width>140</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>8</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Frame Rate</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_4">
   <property name="geometry">
    <rect>
     <x>310</x>
     <y>130</y>
     <width>41</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="text">
    <string>×</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label_5">
   <property name="geometry">
    <rect>
     <x>310</x>
     <y>70</y>
     <width>41</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="text">
    <string>×</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label_7">
   <property name="geometry">
    <rect>
     <x>510</x>
     <y>70</y>
     <width>31</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="text">
    <string>px</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label_8">
   <property name="geometry">
    <rect>
     <x>510</x>
     <y>130</y>
     <width>31</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="text">
    <string>px</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label_9">
   <property name="geometry">
    <rect>
     <x>360</x>
     <y>190</y>
     <width>31</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="text">
    <string>fps</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_10">
   <property name="geometry">
    <rect>
     <x>360</x>
     <y>240</y>
     <width>31</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="text">
    <string>sec</string>
   </property>
  </widget>
  <widget class="Line" name="line">
   <property name="geometry">
    <rect>
     <x>390</x>
     <y>200</y>
     <width>11</width>
     <height>20</height>
    </rect>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>2</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="Line" name="line_2">
   <property name="geometry">
    <rect>
     <x>390</x>
     <y>250</y>
     <width>11</width>
     <height>20</height>
    </rect>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>2</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="Line" name="line_3">
   <property name="geometry">
    <rect>
     <x>390</x>
     <y>210</y>
     <width>20</width>
     <height>50</height>
    </rect>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="lineWidth">
    <number>1</number>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
  </widget>
  <widget class="QFrame" name="frame">
   <property name="geometry">
    <rect>
     <x>549</x>
     <y>20</y>
     <width>231</width>
     <height>181</height>
    </rect>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Raised</enum>
   </property>
   <widget class="QFrame" name="frameImage">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>40</y>
      <width>131</width>
      <height>101</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
   </widget>
   <widget class="QWidget" name="passepartoutRight" native="true">
    <property name="geometry">
     <rect>
      <x>181</x>
      <y>0</y>
      <width>50</width>
      <height>181</height>
     </rect>
    </property>
   </widget>
   <widget class="QWidget" name="passepartoutLeft" native="true">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>50</width>
      <height>181</height>
     </rect>
    </property>
   </widget>
   <widget class="QWidget" name="passepartoutTop" native="true">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>0</y>
      <width>131</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QWidget" name="passepartoutBottom" native="true">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>140</y>
      <width>131</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <zorder>passepartoutBottom</zorder>
   <zorder>passepartoutTop</zorder>
   <zorder>passepartoutRight</zorder>
   <zorder>passepartoutLeft</zorder>
   <zorder>frameImage</zorder>
  </widget>
  <widget class="QPushButton" name="cmdMax_2">
   <property name="geometry">
    <rect>
     <x>620</x>
     <y>200</y>
     <width>181</width>
     <height>81</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Recenter</string>
   </property>
  </widget>
  <widget class="QPushButton" name="cmdExpMax">
   <property name="geometry">
    <rect>
     <x>400</x>
     <y>280</y>
     <width>121</width>
     <height>81</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Max</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_11">
   <property name="geometry">
    <rect>
     <x>360</x>
     <y>300</y>
     <width>31</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="text">
    <string>sec</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_12">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>300</y>
     <width>140</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>8</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Exposure</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_13">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>360</y>
     <width>140</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>8</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Analog Gain</string>
   </property>
  </widget>
  <widget class="QComboBox" name="comboGain">
   <property name="geometry">
    <rect>
     <x>159</x>
     <y>359</y>
     <width>191</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>6</pointsize>
     <weight>50</weight>
     <italic>false</italic>
     <bold>false</bold>
    </font>
   </property>
   <item>
    <property name="text">
     <string>12dB (x4)</string>
    </property>
   </item>
  </widget>
  <widget class="QPushButton" name="cmdOK">
   <property name="geometry">
    <rect>
     <x>620</x>
     <y>400</y>
     <width>181</width>
     <height>81</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Done</string>
   </property>
  </widget>
  <widget class="QPushButton" name="cmdRecMode">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>400</y>
     <width>181</width>
     <height>81</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Record Modes</string>
   </property>
  </widget>
  <widget class="QWidget" name="widget" native="true">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>550</width>
     <height>481</height>
    </rect>
   </property>
  </widget>
  <widget class="QWidget" name="widget_2" native="true">
   <property name="geometry">
    <rect>
     <x>550</x>
     <y>0</y>
     <width>251</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QWidget" name="widget_3" native="true">
   <property name="geometry">
    <rect>
     <x>550</x>
     <y>200</y>
     <width>251</width>
     <height>281</height>
    </rect>
   </property>
  </widget>
  <widget class="QWidget" name="widget_4" native="true">
   <property name="geometry">
    <rect>
     <x>780</x>
     <y>20</y>
     <width>21</width>
     <height>181</height>
    </rect>
   </property>
  </widget>
  <widget class="CamLineEdit" name="lineRate">
   <property name="geometry">
    <rect>
     <x>150</x>
     <y>180</y>
     <width>211</width>
     <height>61</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="text">
    <string>999.990 000 1</string>
   </property>
  </widget>
  <widget class="CamLineEdit" name="linePeriod">
   <property name="geometry">
    <rect>
     <x>150</x>
     <y>230</y>
     <width>211</width>
     <height>61</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="text">
    <string>4 000.000 01</string>
   </property>
  </widget>
  <widget class="CamLineEdit" name="lineExp">
   <property name="geometry">
    <rect>
     <x>150</x>
     <y>290</y>
     <width>211</width>
     <height>61</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>6</pointsize>
    </font>
   </property>
   <property name="text">
    <string>990.00u</string>
   </property>
  </widget>
  <zorder>widget_2</zorder>
  <zorder>widget_3</zorder>
  <zorder>widget_4</zorder>
  <zorder>widget</zorder>
  <zorder>label</zorder>
  <zorder>comboRes</zorder>
  <zorder>spinHRes</zorder>
  <zorder>spinVRes</zorder>
  <zorder>spinHOffset</zorder>
  <zorder>spinVOffset</zorder>
  <zorder>label_2</zorder>
  <zorder>cmdMax</zorder>
  <zorder>label_3</zorder>
  <zorder>label_4</zorder>
  <zorder>label_5</zorder>
  <zorder>label_7</zorder>
  <zorder>label_8</zorder>
  <zorder>label_9</zorder>
  <zorder>label_10</zorder>
  <zorder>line</zorder>
  <zorder>line_2</zorder>
  <zorder>line_3</zorder>
  <zorder>cmdMax_2</zorder>
  <zorder>cmdExpMax</zorder>
  <zorder>label_11</zorder>
  <zorder>label_12</zorder>
  <zorder>label_13</zorder>
  <zorder>comboGain</zorder>
  <zorder>cmdOK</zorder>
  <zorder>cmdRecMode</zorder>
  <zorder>lineRate</zorder>
  <zorder>linePeriod</zorder>
  <zorder>lineExp</zorder>
  <zorder>frame</zorder>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CamSpinBox</class>
   <extends>QSpinBox</extends>
   <header>camspinbox.h</header>
  </customwidget>
  <customwidget>
   <class>CamLineEdit</class>
   <extends>QLineEdit</extends>
   <header>camLineEdit.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
